src/crystal/system/win32/file_descriptor.cr

```diff
@@ -276,6 +277,59 @@ module Crystal::System::FileDescriptor
   end
 end
 
+private module ConsoleUtils
+  # N UTF-16 code units correspond to no more than 3*N UTF-8 code units.
+  private BUFFER_SIZE = 64
+  @@utf16_buffer = Slice(UInt16).new(BUFFER_SIZE)
```

Comment:
If `@@utf16_buffer` never contains incomplete code unit sequences, then its state doesn't need to be persistent, only `@@utf8_buffer` needs to. Therefore `utf16_buffer = uninitialized UInt16[BUFFER_SIZE]` inside `.fill_buffer` should suffice and there is no need for a class variable here. (Even if persistence is needed, only a single `UInt16` is required to keep track of the previous `.fill_buffer`'s decoding state)